### PR TITLE
Automatically select a valid MatchType for the given country code

### PR DESCRIPTION
**Pull request by hmac**
Not all match types are valid for each country. Instead of always using
`MatchBlock`, select the a match type which is valid for the given
country. In a future change we'll also allow users to specify a match
type.

### CODE REVIEW

File: lib/creditsafe/match_type.rb [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      MX: [CLOSEST_KEYWORDS],
+      DK: [MATCH_BEGINNING],
+      AU: [MATCH_BEGINNING],
+      CN: [CLOSEST_KEYWORDS],
+      IN: [CLOSEST_KEYWORDS],
+      BD: [CLOSEST_KEYWORDS],
+      LK: [CLOSEST_KEYWORDS],
+      PK: [CLOSEST_KEYWORDS],
+      NP: [CLOSEST_KEYWORDS],
+      PLC: [CLOSEST_KEYWORDS],
```
Comment (hmac) [inline lib/creditsafe/match_type.rb]: I'm not convinced this is a real country...

File: lib/creditsafe/match_type.rb [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      BY: [MATCH_BLOCK_OR_WORDS],
+      GE: [MATCH_BLOCK_OR_WORDS],
+      KZ: [MATCH_BLOCK_OR_WORDS],
+      UZ: [MATCH_BLOCK_OR_WORDS],
+      TJ: [MATCH_BLOCK_OR_WORDS],
+      TM: [MATCH_BLOCK_OR_WORDS],
+      KG: [MATCH_BLOCK_OR_WORDS],
+      KM: [MATCH_BLOCK_OR_WORDS],
+      HK: [MATCH_BLOCK_OR_WORDS],
+      AT: [MATCH_BLOCK_OR_WORDS],
```
Comment (keylorch) [inline lib/creditsafe/match_type.rb]: AT should be `MatchWords`

File: lib/creditsafe/match_type.rb [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+      IT: [CLOSEST_KEYWORDS],
+      BT: [CLOSEST_KEYWORDS],
```
Comment (keylorch) [inline lib/creditsafe/match_type.rb]: I think this should be `BR` (I'm comparing with Creditsafe's spreadsheet)

File: lib/creditsafe/match_type.rb [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+      HU: [MATCH_BEGINNING],
+      TW: [MATCH_BEGINNING],
+      KR: [MATCH_BLOCK_OR_WORDS],
+      FI: [CLOSEST_KEYWORDS],
+      MX: [CLOSEST_KEYWORDS],
+      DK: [MATCH_BEGINNING],
+      AU: [MATCH_BEGINNING],
```
Comment (keylorch) [inline lib/creditsafe/match_type.rb]: AU should be `ClosestKeywords`

File: lib/creditsafe/match_type.rb (hunk 3 above)
Comment (hmac) [reply lib/creditsafe/match_type.rb]: Yep!

File: lib/creditsafe/match_type.rb (hunk 4 above)
Comment (hmac) [reply lib/creditsafe/match_type.rb]: 🙏

**Comment by hmac** [comment]
Thanks for picking up on those! I obviously wasn't paying close enough attention when I wrote this!

**Pull request merged**